specification/maps/data-plane/Route/preview/2024-04-01-preview/route.json

```diff
@@ -0,0 +1,2097 @@
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OptionalId"
+        },
+        {
+          "$ref": "#/definitions/DirectionsRequest"
+        }
+      ]
+    },
+    "DirectionsBatchRequestBody": {
+      "description": "The list of directions queries/requests to process. The list can contain a max of 100 queries and must contain at least 1 query.",
+      "type": "object",
+      "properties": {
+        "batchItems": {
+          "description": "The list of queries to process.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DirectionsBatchRequestItem"
+          }
+        }
+      }
+    },
+    "DirectionsBatchResponseItem": {
+      "description": "Batch response item",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OptionalId"
+        },
+        {
+          "$ref": "#/definitions/RouteDirectionsResponse"
+        },
+        {
+          "type": "object",
+          "description": "error details",
+          "properties": {
+            "error": {
+              "$ref": "../../../../../common-types/data-plane/v1/types.json#/definitions/ErrorDetail"
+            }
+          }
+        }
+      ]
+    },
+    "DirectionsBatchResponseSummary": {
+      "description": "Summary for the batch request",
+      "type": "object",
+      "properties": {
+        "successfulRequests": {
+          "description": "Number of successful requests in the batch",
+          "type": "integer",
+          "format": "int32"
+        },
+        "totalRequests": {
+          "description": "Total number of requests in the batch",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "DirectionsBatchResponse": {
+      "description": "This object is returned from a successful Directions Batch service call.",
+      "type": "object",
+      "properties": {
+        "summary": {
+          "$ref": "#/definitions/DirectionsBatchResponseSummary"
+        },
+        "batchItems": {
+          "description": "Array containing the batch results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DirectionsBatchResponseItem"
+          }
+        },
+        "nextLink": {
+          "$ref": "#/definitions/NextLink"
+        }
+      }
+    },
+    "AsyncBatchRequestBody": {
+      "type": "object",
+      "description": "Universal asynchronous batch request body.",
+      "properties": {
+        "inputBlobUrl": {
+          "description": "The URL to the input blob containing the batch of queries. This should be a valid URL pointing to a blob of an Azure Storage Account",
+          "type": "string",
+          "format": "uri"
+        },
+        "outputStorageUrl": {
+          "description": "The URL that points to a blob storage where the batch results will be stored.",
+          "type": "string",
+          "format": "uri"
+        },
```

Comment:
Is this a storage account endpoint? Does your service access it and create the container and blobs?

Reply:
It's a storage account endpoint. e.g. `https://uditest.blob.core.windows.net`

Reply:
I am wondering how user know which file in that `azure-maps-async-batch` container is the result of this e.g. route, especially when user do this async multiple times / for multiple routes, and give the same storage account endpoint `outputStorageUrl`.

Reply:
In the Get Operation API, it has a field, `outputBlobUrl`, which points to the blob of the final result.

Reply:
Is there any design consideration that we have `outputStorageUrl` points to a storage account endpoint, instead of a e.g. `outputStorageBlobUrl` points to a single blob?

From the response and `outputBlobUrl`, it appears only output to 1 file. So, it seems natural for the API to let user specify the output file/blob location.

Reply:
There are additional intermediate files that will also reside in the outputStorageUrl. Given our service's Bring Your Own Storage (BYOS) approach, we ensure that all files generated from customer input are stored in the storage provided by the customer, thereby allowing them full control over their data.